action/protocol/account/protocol.go

```diff
@@ -43,9 +42,6 @@ func (p *Protocol) Handle(ctx context.Context, act action.Action, sm protocol.St
 
 // Validate validates an account
 func (p *Protocol) Validate(ctx context.Context, act action.Action) error {
-	p.mu.Lock()
```

Comment:
What happens if Protocol object gets mutated while validating?

Reply:
mutation should happen after validating. One step back, mutation shouldn't affect reading from the committed data in the db.